File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        HashSet<Character> set = new HashSet<>();

        int l = 0;
        int r = 0;
        int max = 0;

        while(r < s.length()){
            if (!set.contains(s.charAt(r))){
                set.add(s.charAt(r));
                max = Math.max(max, set.size());
                r++;
            }
            else{
                set.remove(s.charAt(l));
                l++;
            }
        }


        return max;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp  = [1] * len(nums)
        n = len(nums)

        for i in range(n - 1, -1, -1):
            for j in range(i, n):
                if nums[i] < nums[j]:
                    dp[i] = max(dp[i], 1 + dp[j])
        
        return max(dp)
        